File: src/main/java/pl/mjedynak/idea/plugins/builder/gui/ChooserDisplayerActionListener.java
package pl.mjedynak.idea.plugins.builder.gui;

import com.intellij.codeInsight.CodeInsightBundle;
import com.intellij.ide.util.PackageChooserDialog;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiPackage;
import com.intellij.ui.ReferenceEditorComboWithBrowseButton;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import org.jetbrains.annotations.NotNull;
import pl.mjedynak.idea.plugins.builder.factory.PackageChooserDialogFactory;

public class ChooserDisplayerActionListener implements ActionListener {

    private final @NotNull ReferenceEditorComboWithBrowseButton comboWithBrowseButton;
    private final @NotNull Project project;

    public ChooserDisplayerActionListener(
            @NotNull ReferenceEditorComboWithBrowseButton comboWithBrowseButton, @NotNull Project project) {
        this.comboWithBrowseButton = comboWithBrowseButton;
        this.project = project;
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        PackageChooserDialog chooser = PackageChooserDialogFactory.getPackageChooserDialog(
                CodeInsightBundle.message("dialog.create.class.package.chooser.title"), project);
        chooser.selectPackage(comboWithBrowseButton.getText());
        chooser.show();
        PsiPackage aPackage = chooser.getSelectedPackage();
        if (aPackage != null) {
            comboWithBrowseButton.setText(aPackage.getQualifiedName());
        }
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/gui/ChooserDisplayerActionListener.kt
package io.github.kingg22.kobra.builder.gui

import com.intellij.codeInsight.CodeInsightBundle
import com.intellij.ide.util.PackageChooserDialog
import com.intellij.openapi.project.Project
import com.intellij.ui.ReferenceEditorComboWithBrowseButton
import io.github.kingg22.kobra.builder.factory.PackageChooserDialogFactory
import java.awt.event.ActionEvent
import java.awt.event.ActionListener

public class ChooserDisplayerActionListener(
    private val comboWithBrowseButton: ReferenceEditorComboWithBrowseButton,
    private val project: Project,
) : ActionListener {
    override fun actionPerformed(e: ActionEvent?) {
        val chooser: PackageChooserDialog = PackageChooserDialogFactory.getPackageChooserDialog(
            CodeInsightBundle.message("dialog.create.class.package.chooser.title"),
            project,
        )
        chooser.selectPackage(comboWithBrowseButton.text)
        chooser.show()
        val aPackage = chooser.getSelectedPackage()
        if (aPackage != null) {
            comboWithBrowseButton.text = aPackage.qualifiedName
        }
    }
}
